Crypto/AsyncCrypto.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include <vector>
+#include <memory>
+#include <string>
+
+#include <openssl/pem.h>    // engine interface
+#include <openssl/aes.h>    // AES_BLOCK_SIZE
+#include <openssl/rand.h>   //generate arrays of random bytes
+
+#include "..\CPPLogger\CPPLogger.h"
+
+#ifndef ASYNC_CRYPTO_H
+#define ASYNC_CRYPTO_H
+
+#define RSA_KEY_LEN 2048
+#define PUBLIC_KEYS_AMOUNT 1
+
+class exceptionAsyncCrypto
+{
+public:
+    exceptionAsyncCrypto(const std::string& msg) noexcept
+    {
+        whatStr = msg;
+    }
+    const std::string what()
+    {
+        return whatStr;
+    }
+private:
+    std::string whatStr = "";
```

Comment:
Private class members should be named like this to follow our code convention:
m_what_str;